### PR TITLE
feat(t02): config redux-toolkit and show 100 jobs

### PR DESCRIPTION
**Pull request by HoangTrieu2803**

### CODE REVIEW

File: src/containers/Home/Jobs/style.scss [hunk 1]
```diff
@@ -0,0 +1,46 @@
+.job-active{
```
Comment (nqbinh13) [inline src/containers/Home/Jobs/style.scss:1]: what is this class style for?

File: src/containers/Home/Jobs/style.scss [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    border: solid 1px red;
+}
+.job{
+    margin-left: 20%;
+    margin-right: 20%;
+    padding: 40px;
+    position: relative;
+    .job__tittle{
```
Comment (nqbinh13) [inline src/containers/Home/Jobs/style.scss:9]: you can use `&__tittle` for the shorter syntax

File: src/containers/Home/Jobs/style.scss [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    right: 40px;
+    background-color: orange;
+    color: white;
+    padding: 2px 10px;
+}
+.pagination{
+    justify-content: center;
+    padding-bottom: 20px;
+}
+.job:hover{
```
Comment (nqbinh13) [inline src/containers/Home/Jobs/style.scss:43]: the same above

File: src/containers/Redux/jobs/jobSlice.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    async(data,thunkApi) =>{
+        try{
+            const data = await api.get("/products")
+            return data.data;
+        }catch(err : any){
+            return thunkApi.rejectWithValue(err.message);
+        }
+    }
+)
+interface JobState {
```
Comment (nqbinh13) [inline src/containers/Redux/jobs/jobSlice.ts:15]: should define this in the `types` file

File: src/containers/components/Navbar/style.css
Comment (nqbinh13) [inline src/containers/components/Navbar/style.css:1]: convert to `.scss` file

File: src/containers/Home/Jobs/ListJob/index.tsx [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+import { RootState } from '../../../Redux/store/store';
+export default function ListJob() {
+    const data = useSelector ((state:RootState) => state.jobs)
+    const dispatch = useDispatch();
+    useEffect(()=>{
+       dispatch(getJobs());
+    },[]);
+    const jobs= data.data;
+    const loading = data.loading;
+    if(loading){
```
Comment (nqbinh13) [inline src/containers/Home/Jobs/ListJob/index.tsx:14]: don't do like this.
Check the render Element into the return 
It looks like: 
`return (<>{loading ? <div>Loading</div> : <div>Some code here</div>}</>)`

File: src/containers/Home/Jobs/ListJob/index.tsx [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+        return(
+            <div>Loading...</div>
+        )
+    }else{
+        return <>
+    {jobs?.map((item: Job, key: number) => {
+        return (      
+            <a key={key} >   
+                <li className={`list-group-item d-flex job`} data-testid="test100Jobs">
+                    <img className='job__img' alt='logo' src={item.img} />
```
Comment (nqbinh13) [inline src/containers/Home/Jobs/ListJob/index.tsx:24]: non-blocker, I see now you just have 10 imgs for first 10 jobs. `mockapi` support to generate the random img, you should try this (nice to have)

File: src/containers/Redux/jobs/jobSlice.ts [hunk 7]
```diff
@@ -0,0 +1,44 @@
+
+import {createSlice,createAsyncThunk,PayloadAction} from "@reduxjs/toolkit"
+import Job from "../../model/job";
+import api from "../../util/util";
+export const getJobs = createAsyncThunk("jobs/getJobs",
```
Comment (nqbinh13) [inline src/containers/Redux/jobs/jobSlice.ts:5]: define type for this

File: src/containers/Home/Jobs/ListJob/index.tsx [hunk 8]
```diff
@@ -0,0 +1,45 @@
+import React,{useEffect,useState} from 'react'
+import Job from '../../../model/job';
+import { getJobs } from '../../../Redux/jobs/jobSlice';
+import {useSelector , useDispatch} from "react-redux"
+import { RootState } from '../../../Redux/store/store';
+export default function ListJob() {
+    const data = useSelector ((state:RootState) => state.jobs)
```
Comment (nqbinh13) [inline src/containers/Home/Jobs/ListJob/index.tsx:7]: you can define like: 
`const {data: jobs, loading} = useSelector((state:RootState) => state.jobs)`
so you don't need lines 12, 13

File: src/containers/Home/Jobs/ListJob/index.tsx [hunk 9]
```diff
@@ -0,0 +1,45 @@
+import React,{useEffect,useState} from 'react'
```
Comment (nqbinh13) [inline src/containers/Home/Jobs/ListJob/index.tsx:1]: remove redundant import

File: src/containers/Home/Jobs/style.scss (hunk 1 above)
Comment (HoangTrieu2803) [reply src/containers/Home/Jobs/style.scss:1]: i forgot remove it. I've just did it

File: src/containers/Home/Jobs/style.scss (hunk 2 above)
Comment (HoangTrieu2803) [reply src/containers/Home/Jobs/style.scss:9]: i've changed

File: src/containers/Home/Jobs/style.scss [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+            color: #06c;
+        }
+    }
+    .job__img{
+        width: 60px;
+        height: 60px;
+        border: 1px black solid;
+    }
+}
+.job::after{
```
[the remark this answers is not in the document]
Comment (HoangTrieu2803) [reply src/containers/Home/Jobs/style.scss:30]: i've moved

File: src/containers/Redux/jobs/jobSlice.ts (hunk 4 above)
Comment (HoangTrieu2803) [reply src/containers/Redux/jobs/jobSlice.ts:15]: i've defined

File: src/containers/components/Navbar/style.css
Comment (HoangTrieu2803) [reply src/containers/components/Navbar/style.css:1]: i've converted

File: src/containers/Home/Jobs/ListJob/index.tsx (hunk 5 above)
Comment (HoangTrieu2803) [reply src/containers/Home/Jobs/ListJob/index.tsx:14]: i've changed

File: src/containers/Home/Jobs/ListJob/index.tsx [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+    const loading = data.loading;
+    if(loading){
+        return(
+            <div>Loading...</div>
+        )
+    }else{
+        return <>
+    {jobs?.map((item: Job, key: number) => {
+        return (      
+            <a key={key} >
```
[the remark this answers is not in the document]
Comment (HoangTrieu2803) [reply src/containers/Home/Jobs/ListJob/index.tsx:22]: i've solved

File: src/containers/Home/Jobs/ListJob/index.tsx (hunk 6 above)
Comment (HoangTrieu2803) [reply src/containers/Home/Jobs/ListJob/index.tsx:24]: i've tried it

File: src/containers/Home/Jobs/ListJob/index.tsx (hunk 9 above)
Comment (HoangTrieu2803) [reply src/containers/Home/Jobs/ListJob/index.tsx:1]: i've removed

File: src/containers/Home/Jobs/ListJob/index.tsx (hunk 8 above)
Comment (HoangTrieu2803) [reply src/containers/Home/Jobs/ListJob/index.tsx:7]: i've changed

File: src/containers/Home/Jobs/style.scss [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+    }
+    &:hover{
+        box-shadow: 0 0 5px 0 rgb(212, 16, 16)
+    }
+    &__tittle{
+        margin-left: 50px;
+        .job__tech{
+            text-align: left;
+            position: relative;
+            &::after{
```
Comment (nqbinh13) [inline src/containers/Home/Jobs/style.scss:25]: shoudn't create link by pseudo-class

File: src/containers/Home/Jobs/ListJob/index.tsx [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+    return (
+        <div>
+            <h2 data-testid="test100Job">{jobs.length} Jobs For "Chất" Developers</h2>
+            {loading ? <Loading /> : jobs?.map((item: Job, key: number) => {
+                return (
+                    <li key={key} className={`list-group-item d-flex job`}>
+                        <img className='job__img' alt='logo' src={item.img} />
+                        <div className='job__tittle'>
+                            <h4>{item.jobName}</h4>
+                            <p>
```
Comment (nqbinh13) [inline src/containers/Home/Jobs/ListJob/index.tsx:22]: don't need `p` element here

Comment (HoangTrieu2803) [reply src/containers/Home/Jobs/ListJob/index.tsx]: i've removed it

File: src/containers/Home/Jobs/style.scss (hunk 12 above)
Comment (HoangTrieu2803) [reply src/containers/Home/Jobs/style.scss]: i've changed

**Pull request merged**